game-core/src/main/java/games/strategy/engine/pbem/NodeBbForumPoster.java

```diff
@@ -82,7 +82,9 @@ private void post(
     try (CloseableHttpResponse response = client.execute(post)) {
       final int code = response.getStatusLine().getStatusCode();
       if (code != HttpURLConnection.HTTP_OK) {
-        throw new IllegalStateException("Forum responded with code " + code);
+        throw new IllegalStateException(String.format(
+            "Forum responded with code %s and message '%s'",
+            code, EntityUtils.toString(response.getEntity())));
```

Comment:
Will entity always be non-null? Empty and error responses are certain to be not null?

Reply:
Does `response.toString()` provide too much info?

Reply:
I don't think `response.toString()` does provide any information. I might be wrong, but `EntityUtils` is what's used everywhere so I used the thing I know works.

The documentation is lacking a lot of details like if certain values can be null, but AFAIK `getEntity()` always returned a non-null value. If anything goes wrong with the request that might prevent creating an entity this will result in an exception and never reach this point of execution.

Reply:
Looking through source code implementations, it looked like at least one implementation of `Response` did have a `toString` implementation and the other delegated back to that same implementation.

I don't think we can assume a non-null response.